### PR TITLE
Add push to nomis capability. Send approved/rejected on decision from DM

### PR DESCRIPTION
**Pull request by whitfield-mj**

### CODE REVIEW

File: server/routes/approval.js [hunk 1]
```diff
@@ -20,7 +18,7 @@ module.exports = ({licenceService, prisonerService}) => (router, audited) => {
 
             const {nextPath, pageDataMap} = formConfig[formName];
             const dataPath = pageDataMap || ['licence', 'approval', 'release'];
-            const data = getIn(res.locals.licence, dataPath) || {};
+            const data = firstItem(req.flash('userInput')) || getIn(res.locals.licence, dataPath) || {};
```
Comment (petergphillips) [inline server/routes/approval.js]: what's going on here?

File: server/services/nomisPushService.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+const logger = require('../../log.js');
+
+module.exports = nomisClientBuilder => {
+
+    async function pushStatus(bookingId, approvalDecision, systemToken) {
+        const nomisClient = nomisClientBuilder(systemToken);
+        const status = (() => {
```
Comment (petergphillips) [inline server/services/nomisPushService.js]: don't really understand this.  You creating a function to work out the status and then calling the function immediately?  Why not just have a case statement, why the need to wrap in a dynamic function?

File: test/supertestSetup.js [hunk 3]
```diff
@@ -19,11 +19,11 @@ const auditStub = {
     record: sinon.stub()
 };
 
-const signInServiceStub = {
+const createSignInServiceStub = () => ({
```
Comment (petergphillips) [inline test/supertestSetup.js]: why the change?

File: server/routes/approval.js (hunk 1 above)
Comment (whitfield-mj) [reply server/routes/approval.js]: If the page is redirected to after a validation error then the data to fill the form fields with will be in req.flash. If it is accessed having already been inputted it will be in the licence. Otherwise empty

File: server/services/nomisPushService.js (hunk 2 above)
Comment (whitfield-mj) [reply server/services/nomisPushService.js]: Just another way of doing it. Not uncommon to have a self invoking function in JS. Happy to use Switch if preferred

File: test/supertestSetup.js (hunk 3 above)
Comment (whitfield-mj) [reply test/supertestSetup.js]: Most of the others now use the create method. Having one that is used across tests caused the test suite to hang when I called a reset

File: server/services/nomisPushService.js (hunk 2 above)
Comment (petergphillips) [reply server/services/nomisPushService.js]: I would prefer:
```javascript
        const bob = {
            Yes: 'Approved',
            No: 'Rejected'
        }
        const status = bob[approvalDecision]
```

File: test/supertestSetup.js (hunk 3 above)
Comment (petergphillips) [reply test/supertestSetup.js]: :+1:
